crowbar_framework/app/models/service_object.rb

```diff
@@ -1576,4 +1616,35 @@ def profile(name, &block)
       block.call
     end
   end
+
+  def equal_or_match_version(value, maybe_regexp)
+    to_version = lambda do |version|
+      case version.to_s
+      when /^(\d+)\.(\d+)\.(\d+)$/
+        [ $1.to_i, $2.to_i, $3.to_i ]
+      when /^(\d+)\.(\d+)$/
+        [ $1.to_i, $2.to_i, 0 ]
+      when /^(\d+)$/
+        [ $1.to_i, 0, 0 ]
+      else
+        [0, 0, 0]
+      end
+    end
+
+    cmp = lambda do |op, version|
+      (to_version.call(value).zip to_version.call(version)).all? do |v1, v2|
+        v1.send(op, v2)
+      end
+    end
+
+    op_value = maybe_regexp.scan(/^(>=?|<=?)\s*([.\d]+)$/)
```

Comment:
For the record, I like the code flow of the js bits a bit better (where this regexp is only used when we're not in the `/foo/` case.